docs/TextRegion.md

```diff
@@ -0,0 +1,86 @@
+* align (Align/str, optional) - the horizontal alignment of the text in the region. (Default: Align.L)
+* line_height (float, optional) - This is a factor by which the line spacing will be different from the font height. It works similar to the attribute of the same name in HTML/CSS. (default: 1.0)
+* print_sh (bool, optional) - Treat a soft-hyphen (\\u00ad) as a printable character, instead of a line breaking opportunity. (Default: False)
+* skip_leading_spaces (default: False) - This flag is primarily used by `write_html()`, but may also have other uses. It removes all space characters at the beginning of each line.
+* wrapmode (default "WORD") - 
+
+All of those values can be overriden for each individual paragraph.
+
+
+### Common methods ###
+
+* paragraph() [see characteristics parameters below] - establish a new paragraph in the text. The text added to this paragraph will start on a new line.
+* write(text: str, link: = None) - write text to the region. This is only permitted when no explicit paragraph is currently active.
+* ln(h: float = None) - Start a new line moving either by the current font height or by the parameter "h". Only permitted when no explicit paragraph is currently active.
+* render() - if the region is not used as a context manager with "with", this method must be called to actually process the added text.
+
+
+## Paragraphs ##
+
+The primary purpose of paragraphs is to enable variations in horizontal text alignment, while the horizontal extents of the text are managed by the text region. To set the alignment, you can use the `align` argument when creating the paragraph, with the same `Align` values as elsewhere in the library.
+
+Note that the `write()` methods of paragraphs and text regions in general don't accept any other argument than "text" and "link".
+
+For more typographical control, you can use the following arguments. Most of those override the settings of the current region when set, and default to the value set there.
+
+* align (Align, optional) - The horizontal alignment of the paragraph.
+* line_height (float, optional) - factor by which the line spacing will be different from the font height. (default: by region) 
+* top_margin (float, optional) -  how much spacing is added above the paragraph. No spacing will be added at the top if the paragraph if the current y position is at (or above) the top margin of the page. (Default: 0.0)
+* bottom_margin (float, optional) - Those two values determine how much spacing is added below the paragraph. No spacing will be added at the bottom if it would result in overstepping the bottom margin of the page. (Default: 0.0)
+* skip_leading_spaces (float, optional) - removes all space characters at the beginning of each line.
+* wrapmode (WrapMode, optional)
+
+Other than text regions, paragraphs should alway be used as context managers and never be reused. Violating those rules may result in the entered text turning up on the page out of sequence.
```

Comment:
```suggestion
Other than text regions, paragraphs should always be used as context managers and never be reused. Violating those rules may result in the entered text turning up on the page out of sequence.
```
I do not completely understand the first sentence, sorry 😅

What do you mean by "Other than text regions"?

Reply:
Further up it is explained that a region may be used as a context manager several times, with other content added in between, and fed with new text each time.

In contrast to that, each paragraph has its fixed position in a sequence, and reopening it as a context manager would cause chaos. The reason is that its content doesn't get rendered immediately after closing, but only after the parent region gets closed.

Reply:
OK. This sentence may need some reformulation to become a bit clearer 😊

And by the way, it seems to me that instead of warning users about this in our documentation,
it would be relatively easy and more efficient to add some basic check in `Paragraph.__enter__()` about this,
in order to raise an error if is used incorrectly. Maybe simply a check that `self._region._active_paragraph is not None`?